lib/checks/canFindRoots.js

```diff
@@ -0,0 +1,33 @@
+const Node = require('../node');
+const resolvesToConstant = require('./resolvesToConstant.js');
+
+function canFindRoots(equation) {
+
+  // Equation should be in the form of factor * factor = 0, factor^power = 0, or identifier = 0
+  // e.g (x - 2)^2 = 0, x(x + 2)(x - 2) = 0, x = 0
+  const left = equation.leftNode;
+  const right = equation.rightNode;
+
+  const zeroRightSide = Node.Type.isConstant(right)
+        && parseFloat(right.value) === 0;
+
+  const hasRoots = Node.Type.isOperator(left, '*') || Node.Type.isOperator(left, '^');
```

Comment:
I feel like this isn't a good name for this variable, because this condition could be true without it having roots

I'd just add this into the if below if you can't think of a better name

though think about this: what if the left side is parens e.g. `((x-2)^2)` will it still work (maybe that gets handled elsewhere)?